### PR TITLE
Add support for ESTScan scores matrices (smat) data type

### PR DESCRIPTION
**Pull request by gregvonkuster**

### CODE REVIEW

File: lib/galaxy/datatypes/text.py [hunk 1]
```diff
@@ -542,3 +542,67 @@ def set_peek( self, dataset, is_multi_byte=False ):
...
+                    # Smat files are not tabular.
+                    return False
+                items = line.split()
+                if len(items) != 4:
+                    return False
+                for item in items:
+                    # Make sure each item is an integer.
+                    if re.match(r"[-+]?\d+$", item) is None:
+                        return False
+        fh.close()
```
Comment (bgruening) [inline lib/galaxy/datatypes/text.py]: If you use `with` the filehandle is closed automatically.

File: lib/galaxy/datatypes/text.py [hunk 2]
```diff
@@ -542,3 +542,67 @@ def set_peek( self, dataset, is_multi_byte=False ):
...
+        False
+        >>> fname = get_test_fname('1.smat')
+        >>> Smat().sniff(fname)
+        True
+        """
+        line_no = 0
+        with open(filename, "r") as fh:
+            line_no += 1
+            if line_no > 10000:
+                return True
```
Comment (bgruening) [inline lib/galaxy/datatypes/text.py:591]: Why this? Do you mean False?

Comment (gregvonkuster) [reply lib/galaxy/datatypes/text.py:591]: The intent is that if 10000 lines have been inspected and it is a smat datatype so far, then the rest of the file is likely smat as well.

File: lib/galaxy/datatypes/text.py (hunk 1 above)
Comment (gregvonkuster) [reply lib/galaxy/datatypes/text.py]: This is outside of the with block.

**Comment by jmchilton** [comment]
Cool beans @gregvonkuster - keep up the great work!

**Comment by gregvonkuster** [comment]
Thanks for the merge!  ;)

**Pull request merged**